studio/test/aem/aem.test.js

```diff
@@ -1,8 +1,106 @@
 import { expect } from '@esm-bundle/chai';
+import sinon from 'sinon';
 import { AEM, filterByTags } from '../../src/aem/aem.js';
 
-describe('aem.js', () => {
-    const aem = new AEM('test');
+describe('AEM SDK', () => {
+    let aem;
+    let fetchStub;
+
+    beforeEach(() => {
+        aem = new AEM('test-bucket', '/test-url');
+        
+        // Mock the private properties correctly
+        Object.defineProperties(aem, {
+            cfFragmentsUrl: {
+                value: '/test-url/content/dam/fragments',
+                writable: true
+            },
+            headers: {
+                value: { Authorization: 'test' },
+                writable: true
+            }
+        });
+
+        // Mock sites property with proper structure
+        aem.sites = {
+            cf: {
+                fragments: {
+                    create: sinon.stub().resolves({ id: 'test-id' }),
+                    save: sinon.stub().resolves({ id: 'test-id' }),
+                    publish: sinon.stub().resolves(true),
+                    unpublish: sinon.stub().resolves(true),
+                    delete: sinon.stub().resolves(true),
+                    search: sinon.stub().resolves([]),
+                    getById: sinon.stub().resolves({ id: 'test-id' })
+                }
+            }
+        };
+
+        fetchStub = sinon.stub(window, 'fetch');
+    });
+
+    afterEach(() => {
+        sinon.restore();
+    });
+
+    describe('fragments.create', () => {
+        it('should create fragment successfully', async () => {
+            const mockResponse = {
+                id: 'test-id',
+                path: '/test/path',
+                status: 'Draft',
+            };
+            
+            aem.sites.cf.fragments.create.resolves(mockResponse);
+
+            const result = await aem.sites.cf.fragments.create({
+                title: 'Test Fragment',
+                model: { id: 'test-model' },
+            }, '/test/parent/path');
+
+            expect(result).to.deep.equal(mockResponse);
+            expect(aem.sites.cf.fragments.create.calledOnce).to.be.true;
+        });
+
+        it('should handle creation errors', async () => {
+            const error = new Error('Bad Request');
+            aem.sites.cf.fragments.create.rejects(error);
+
+            try {
+                await aem.sites.cf.fragments.create({}, '/test/path');
+                expect.fail('Should have thrown an error');
+            } catch (err) {
+                expect(err.message).to.include('Bad Request');
+            }
+        });
+    });
+
+    describe('fragments.publish', () => {
+        it('should publish fragment successfully', async () => {
```

Comment:
to test aem.js, I recommend to test the payload of the fetch requests being send.